parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java

```diff
@@ -231,6 +243,31 @@ private WriteBuilder createContextFunc(
       return this;
     }
 
+    private FileEncryptionProperties createEncryptionProperties(EncryptionKeyMetadata keyMetadata) {
+      Preconditions.checkArgument(keyMetadata != null, "Null key metadata");
+
+      // Native Parquet encryption can be activated only by envelope metadata.
+      // Other metadata types imply flat/generic file encryption
+      if (!(keyMetadata instanceof EnvelopeKeyMetadata)) {
```

Comment:
If we only support one type of key metadata, then we should only accept that key metadata right?

Or maybe this should be a key rather than a wrapper class?

Reply:
Sure. Per the comment above, I'm fine with having only this type of key metadata.